File: api/schemas/user_story.py
from pydantic import BaseModel, UUID4, Field
from typing import List, Optional, Union
from enum import Enum

class PriorityEnum(str, Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"

class UserStoryBase(BaseModel):
    id: UUID4
    project_id : UUID4
    title: str
    description: str
    acceptance_criteria: str
    priority : PriorityEnum
    storyPoints: int = Field(..., gt=0)
    labels : List[str]
    issueType: str
    jiraPush : bool

    class Config:
        from_attributes = True 

class UserStoryCreate(BaseModel):
    project_id : UUID4
    title: str
    description: str
    acceptance_criteria: str
    priority : PriorityEnum
    storyPoints: int = Field(..., gt=0)
    labels : List[str]
    issueType: str
    jiraPush : bool

    class Config:
        from_attributes = True 

class UserStoryUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    acceptance_criteria: Optional[Union[str, List[str]]] = None
    priority : Optional[PriorityEnum] = None
    storyPoints: Optional[int] = Field(None, gt=0)
    labels : Optional[List[str]] = None
    issueType : Optional[str] = None
    jiraPush : Optional[bool] = None

    class Config:
        from_attributes = True 

class UserStoryGenerate(BaseModel):
    project_id: UUID4
    user_prompt: str

File: client/schema/user_story.ts
// TODO: add story points number gt 0 validation
export interface UserStory {
  id: string;
  project_id: string;
  title: string;
  description: string;
  acceptance_criteria: string;
  storyPoints: number;
  priority: string;
  labels: string[];
  issueType: string;
  jiraPush: boolean;
  jira_id: number;
  jira_ignored: boolean;
  created_at: Date;
}

export interface UserStoryCreate {
  project_id: string;
  title: string;
  description?: string;
  acceptance_criteria?: string;
  storyPoints?: number;
  priority?: string;
  labels?: string[];
  jiraPush: boolean;
  issueType: string;
  jira_id: number;
  jira_ignored?: boolean;
}

export interface UserStoryUpdate {
  title?: string;
  description?: string;
  acceptance_criteria?: string;
  storyPoints?: number;
  labels?: string[];
  priority?: string;
  jiraPush?: boolean;
  issueType?: string;
  jira_id?: number;
  jira_ignored?: boolean;
}

export interface UserStoryGenerate {
  project_id: string;
  user_prompt: string;
}
